code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -41,6 +41,23 @@
 	gold_core_spawnable = 0
 
 /mob/living/simple_animal/pet/cat/Life()
+	if(!stat && !buckled)
+		if(prob(1))
+			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail."))

```

Comment:
You should be able to rub the belly exactly twice. On the third rub you get mauled.